### ISSUE TITLE
.aderc missing for ADE installation

### ISSUE THREAD

**Issue by kappa95** [ros2-realtime-demo/pendulum#70]
**Describe the bug**
I am following the guide for [installation with ADE](https://github.com/ros2-realtime-demo/pendulum/blob/foxy/docs/installation.md) but after `ade start --update --enter` I get:
```bash
ERROR: Needs to run from within a directory with a .aderc file or use ``ade --rc path/to/.aderc start``

```
**Desktop:**
 - OS: Ubuntu
 - Version Focal 20.04 LTS

How can I solve?

**Comment by carlossvg** [comment]
@kappa95 Thanks for reporting. I discussed with @LanderU and the problem is that you have to be in the same directory as the .aderc file. This should fix the issue:

```bash
cd ~/adehome/pendulum_ws/src/pendulum/ade
source .aderc
ade start --update --enter
```

This should work too:

```bash
cd ~/adehome/
ade --rc pendulum_ws/src/pendulum/ade/.aderc start --update --enter
```

I will update the instructions adding this missing steps.

**Comment by kappa95** [comment]
Thank you so much for your fast answer and help
I checked also to have followed the whole guide for docker and I have setted it also to work without sudo
I get a newer error: 

```bash
ade-base-foxy: Pulling from ros2realtimedemo/pendulum
Digest: sha256:a6e461d1dd7b7913f76d57715ef71dc3ad659a2a172198d6ef4d1e5bfabdf6f4
Status: Image is up to date for ros2realtimedemo/pendulum:ade-base-foxy
docker.io/ros2realtimedemo/pendulum:ade-base-foxy
Starting ade with the following images:
pendulum | n/a | ade-base-foxy | ros2realtimedemo/pendulum:ade-base-foxy
non-network local connections being added to access control list

Current default time zone: 'Europe/Rome'
Local time is now:      Thu Mar 25 22:35:03 CET 2021.
Universal Time is now:  Thu Mar 25 21:35:03 UTC 2021.

Adding user kappa95 to group video
Adding user kappa95 to group tracing
ADE startup completed.
Entering ade with following images:
pendulum | n/a | ade-base-foxy | ros2realtimedemo/pendulum:ade-base-foxy
OCI runtime exec failed: exec failed: container_linux.go:367: starting container process caused: chdir to cwd ("/root/ros2_foxy") set in config.json failed: permission denied: unknown
ERROR: Command return non-zero exit code (see above): 126
  docker exec -ti --env COLORFGBG --env TERM -u kappa95 ade bash -li


```

**Comment by LanderU** [comment]
@kappa95, thanks a lot for reporting! Could you please tell us where are you trying to run the image? Are you trying to run this on a non x86_64 platform?

**Comment by kappa95** [comment]
> @kappa95, thanks a lot for reporting! Could you please tell us where are you trying to run the image? Are you trying to run this on a non x86_64 platform?

Is it my laptop with Ubuntu 20.04 64bit with Ryzen 2500U and 8GB of RAM

**Comment by LanderU** [comment]
@kappa95, hmm, this is strange. Could you please do the following?

1. Purge the containers related to this project `docker rmi -f ...`
2. Run again the container without sudo

**Comment by kappa95** [comment]
> @kappa95, hmm, this is strange. Could you please do the following?
> 
>     1. Purge the containers related to this project `docker rmi -f ...`
> 
>     2. Run again the container without sudo

Ok I did this:

``` bash
kappa95@kappa95-asus-laptop-x505za:~/adehome$ docker rmi -f ros2realtimedemo/pendulum:ade-base-foxy 
Untagged: ros2realtimedemo/pendulum:ade-base-foxy
Untagged: ros2realtimedemo/pendulum@sha256:a6e461d1dd7b7913f76d57715ef71dc3ad659a2a172198d6ef4d1e5bfabdf6f4
Deleted: sha256:b6e8ff55021de2b995564c119715b7ce04c7697ccff7d67dcabfbaa7f9df9cc7 ```

and later:

``` bash
ade --rc pendulum_ws/src/pendulum/ade/.aderc start --update --enter
INFO: Using non-default pendulum_ws/src/pendulum/ade/.aderc configuration file
ade-base-foxy: Pulling from ros2realtimedemo/pendulum
e6ca3592b144: Already exists 
534a5505201d: Already exists 
990916bd23bb: Already exists 
478568dfb1f6: Already exists 
417893889705: Already exists 
f65bb81578b4: Pull complete 
6a6a19ee9111: Pull complete 
56a5879473f6: Pull complete 
b5c6b0f453d8: Pull complete 
6b29642f42f9: Pull complete 
a81946b255c1: Pull complete 
3cf64c9dda5a: Pull complete 
Digest: sha256:e58a1676ffe530a5192d29a56e2f1b4fdda96cb22ead8f162e85a23c4a18985e
Status: Downloaded newer image for ros2realtimedemo/pendulum:ade-base-foxy
docker.io/ros2realtimedemo/pendulum:ade-base-foxy
Creating volume container for ros2realtimedemo/pendulum:ade-base-foxy
Starting ade with the following images:
pendulum | n/a | ade-base-foxy | ros2realtimedemo/pendulum:ade-base-foxy
non-network local connections being added to access control list

Current default time zone: 'Europe/Rome'
Local time is now:      Sat Mar 27 14:39:35 CET 2021.
Universal Time is now:  Sat Mar 27 13:39:35 UTC 2021.

Adding user kappa95 to group video
Adding user kappa95 to group tracing
ADE startup completed.
Entering ade with following images:
pendulum | n/a | ade-base-foxy | ros2realtimedemo/pendulum:ade-base-foxy
OCI runtime exec failed: exec failed: container_linux.go:367: starting container process caused: chdir to cwd ("/root/ros2_foxy") set in config.json failed: permission denied: unknown
ERROR: Command return non-zero exit code (see above): 126
  docker exec -ti --env COLORFGBG --env TERM -u kappa95 ade bash -li 
```




Looking at the line of the error: `OCI runtime exec failed: exec failed: container_linux.go:367: starting container process caused: chdir to cwd ("/root/ros2_foxy") set in config.json failed: permission denied: unknown` could be possible that is due to the fact that I have installed ros2 through debian packages and so I have the setup files into `/opt/ros/foxy/` or because no one can access to `/root` without being root user?

**Comment by LanderU** [comment]
@kappa95, could you please check if you've something related to this repo on your `~ .docker/config.json`?

**Comment by kappa95** [comment]
> @kappa95, could you please check if you've something related to this repo on your `~ .docker/config.json`?

I checked hidden files and so on, but i don't have that folder

**Comment by kappa95** [comment]
In any case, today I did a clean install of OS and ROS, and I downgrade everything to Ubuntu 18 and ROS Dashing in order to have the same settings of the lab where I'm starting to collaborate. After this setting up, I discovered a fix of why I wasn't see the URDF with RViz2 (damn `locale` :laughing:).
Thank you so much for your time! Probably I will ask you something else related to the RealTime topic in the next future!
